RTR on winagent fails during Github action execution

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.5| Github Action workflow - rtr on winagent| - | - | - |

# Description

During https://github.com/wazuh/wazuh-jenkins#5066 it was found that `rtr (syscheckd, winagent)` Github action is failing, showing the next error


https://github.com/wazuh/wazuh/actions/runs/5005277940/jobs/8969009817?pr=16742
```
<syscheckd>=============== Running Tests       ===============<syscheckd> 

it looks like wine32 is missing, you should install it.
as root, please execute "apt-get install wine[32](https://github.com/wazuh/wazuh/actions/runs/4979035418/jobs/8910127283#step:5:33)"
wine: created the configuration directory '/home/runner/.wine'
0050:err:explorer:initialize_display_settings Failed to query current display settings for L"\\\\.\\DISPLAY1".
0050:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
0050:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
0050:err:ole:apartment_get_local_server_stream Failed: 0x80004002
002c:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
002c:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
0048:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0048:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
0048:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
0048:err:ole:apartment_createwindowifneeded CreateWindow failed with error 0
0048:err:ole:apartment_createwindowifneeded CreateWindow failed with error 14007
0048:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x800736b7
0048:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x800736b7
0048:err:ole:apartment_get_local_server_stream Failed: 0x800736b7
0048:err:ole:start_rpcss Failed to open RpcSs service
0040:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0040:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
0040:err:setupapi:SetupDefaultQueueCallbackW copy error 1812 L"@wineusb.sys,-1" -> L"C:\\windows\\inf\\wineusb.inf"
Could not find Wine Gecko. HTML rendering will be disabled.
it looks like wine32 is missing, you should install it.
as root, please execute "apt-get install wine32"
009c:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
009c:err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
Could not find Wine Gecko. HTML rendering will be disabled.
wine: configuration in L"/home/runner/.wine" has been updated.
it looks like wine32 is missing, you should install it.
as root, please execute "apt-get install wine32"
00e4:err:process:exec_process L"Z:\\home\\runner\\work\\wazuh\\wazuh\\src\\syscheckd\\build\\bin\\fimdb_unit_test.exe" not supported on this system

 [fimdb_unit_test.exe: FAILED] 
    runTests(moduleName=moduleName)
  File "/home/runner/work/wazuh/wazuh/src/ci/run_check.py", line [59](https://github.com/wazuh/wazuh/actions/runs/4979035418/jobs/8910127283#step:5:60)4, in runTests
    raise ValueError(errorString)
ValueError: Error Running test: 123
Error: Process completed with exit code 1.
```

Despite this check is pointing to master, this is failing also on other PRs that target 4.5 like https://github.com/wazuh/wazuh/actions/runs/4979035418/jobs/8910127283

--- a/.github/actions/install_build_deps/action.yml
+++ b/.github/actions/install_build_deps/action.yml
@@ -23,7 +23,7 @@ runs:
       run: |
         if [[ "${{ inputs.target }}" == "winagent" ]]; then
           sudo dpkg --add-architecture i386 && sudo apt-get update
-          sudo apt-get install -y --install-recommends wine-stable
+          sudo apt-get install -y --install-recommends wine-stable wine32
         else
           echo "Skipping wine installation for this target"
         fi

--- a/.github/actions/install_build_deps/action.yml
+++ b/.github/actions/install_build_deps/action.yml
@@ -22,8 +22,12 @@ runs:
       shell: bash
       run: |
         if [[ "${{ inputs.target }}" == "winagent" ]]; then
-          sudo dpkg --add-architecture i386 && sudo apt-get update
-          sudo apt-get install -y --install-recommends wine-stable wine32
+          sudo dpkg --add-architecture i386
+          sudo mkdir -pm755 /etc/apt/keyrings
+          sudo wget -O /etc/apt/keyrings/winehq-archive.key https://dl.winehq.org/wine-builds/winehq.key
+          sudo wget -NP /etc/apt/sources.list.d/ https://dl.winehq.org/wine-builds/ubuntu/dists/jammy/winehq-jammy.sources
+          sudo apt-get update
+          sudo apt-get install -y --allow-downgrades libc6:i386 libgcc-s1:i386 libstdc++6:i386 wine-stable-i386 wine-stable-amd64 winehq-stable
         else
           echo "Skipping wine installation for this target"
         fi

--- a/.github/workflows/rtr_syscheck.yml
+++ b/.github/workflows/rtr_syscheck.yml
@@ -8,6 +8,8 @@ on:
       - "src/syscheckd/**"
       - "src/unit_tests/syscheckd/**"
       - "src/Makefile"
+      - ".github/workflows/rtr_syscheck.yml"
+      - ".github/actions/**"
 
 jobs:
   rtr:

--- a/.github/workflows/rtr_syscheck.yml
+++ b/.github/workflows/rtr_syscheck.yml
@@ -18,7 +18,8 @@ jobs:
           matrix:
               module: [syscheckd]
               target: [agent, winagent]
-    runs-on: ubuntu-latest
+    # We don't use ubuntu-latest because the install_build_deps action adds a fixed repository for Wine
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v3

--- a/.github/workflows/rtr_syscollector.yml
+++ b/.github/workflows/rtr_syscollector.yml
@@ -9,6 +9,8 @@ on:
       - "src/shared_modules/**"
       - "src/wazuh_modules/syscollector/**"
       - "src/Makefile"
+      - ".github/workflows/rtr_syscollector.yml"
+      - ".github/actions/**"
 
 jobs:
   rtr:

--- a/.github/workflows/rtr_syscollector.yml
+++ b/.github/workflows/rtr_syscollector.yml
@@ -19,7 +19,8 @@ jobs:
           matrix:
               module: [wazuh_modules/syscollector, shared_modules/dbsync, shared_modules/rsync, shared_modules/utils, data_provider]
               target: [agent, winagent]
-    runs-on: ubuntu-latest
+    # We don't use ubuntu-latest because the install_build_deps action adds a fixed repository for Wine
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v3

